src/Modules/OrchardCore.Commerce/Endpoints/Api/CheckoutEndpoint.cs

```diff
@@ -22,13 +22,13 @@ public static IEndpointRouteBuilder AddCheckoutAsyncEndpoint(this IEndpointRoute
 
     [Authorize(AuthenticationSchemes = "Api")]
     private static async Task<IResult> CheckoutAsync(
-        string shoppingCartId,
+        string? shoppingCartId,
         IAuthorizationService authorizationService,
         HttpContext httpContext,
         IPaymentService paymentService
         )
     {
-        if (!await authorizationService.AuthorizeAsync(httpContext.User, ApiPermissions.CommerceApi))
+        if (!await authorizationService.AuthorizeAsync(httpContext.User, Payment.Permissions.Checkout))
```

Comment:
I mean check both permissions.

```suggestion
        if (!await authorizationService.AuthorizeAsync(httpContext.User, ApiPermissions.CommerceApi) ||
            !await authorizationService.AuthorizeAsync(httpContext.User, Payment.Permissions.Checkout))
```